src/coreclr/jit/llvm.cpp

```diff
@@ -1175,38 +1201,76 @@ void Llvm::buildCall(GenTree* node)
 
 void Llvm::buildCast(GenTreeCast* cast)
 {
-    var_types castToType = cast->CastToType();
-    if (castToType == TYP_BOOL && cast->TypeIs(TYP_INT) && cast->CastOp()->TypeIs(TYP_INT))
-    {
-        Value* intValue = _builder.CreateZExt(getGenTreeValue(cast->CastOp()), getLlvmTypeForVarType(TYP_INT));
-        mapGenTreeToValue(cast, intValue); // nothing to do except map the source value to the destination GenTree
-    }
-    else if (castToType == TYP_DOUBLE && cast->CastOp()->TypeIs(TYP_FLOAT))
-    {
-        mapGenTreeToValue(cast, _builder.CreateFPCast(getGenTreeValue(cast->CastOp()), getLlvmTypeForVarType(TYP_DOUBLE)));
-    }
-    else if (cast->TypeIs(TYP_LONG) && genActualTypeIsInt(cast->CastOp()))
-    {
-        // Cast pointer to int if necessary.  TODO-LLVM: candidate for lowering?
-        Value* sourceValue = castIfNecessary(getGenTreeValue(cast->CastOp()), getLlvmTypeForVarType(cast->CastOp()->TypeGet()));
+    var_types castFromType  = genActualType(cast->CastOp());
+    var_types castToType    = cast->CastToType();
+    Value*    castFromValue = consumeValue(cast->CastOp(), getLlvmTypeForVarType(castFromType));
+    Value*    castValue     = nullptr;
```

Comment:
Seems a good idea to add & use `castToLlvmType` here too, instead of individual `getLlvmTypeForVarType(castToType)` calls.